### ISSUE TITLE
Can't find the parameter in the constructor when inherit the class with CoffeeScript

### ISSUE THREAD

**Issue by laiso** [itchyny/sjsp#4]
For Example:

``` coffee
class Animal
  constructor: (opts) ->
    @name = opts.name

class Dog extends Animal

o = new Dog({name: 'cinnamon'})
console.log(o.name)
```

> coffee -c a.coffee

``` js
// Generated by CoffeeScript 1.9.3
(function() {
  var Animal, Dog, o,
    extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
    hasProp = {}.hasOwnProperty;

  Animal = (function() {
    function Animal(opts) {
      this.name = opts.name;
    }

    return Animal;

  })();

  Dog = (function(superClass) {
    extend(Dog, superClass);

    function Dog() {
      return Dog.__super__.constructor.apply(this, arguments);
    }

    return Dog;

  })(Animal);

  o = new Dog({
    name: 'cinnamon'
  });

  console.log(o.name);

}).call(this);
```

> sjsp a.js

``` js
  Animal = (function() { var sjsp__state = sjsp__start("a.js",7,24,"anonymous","Animal = (function() {");
    function Animal(opts) { var sjsp__state = sjsp__start("a.js",8,27,"Animal","function Animal(opts) {");
      this.name = opts.name;; sjsp__end(sjsp__state);
    }

    return (function(){ var sjsp__return = Animal; sjsp__end(sjsp__state); return sjsp__return; } ).call(this);; sjsp__end(sjsp__state);

  })();

  Dog = (function(superClass) { var sjsp__state = sjsp__start("a.js",16,31,"anonymous","Dog = (function(superClass) {");
    extend(Dog, superClass);

    function Dog() { var sjsp__state = sjsp__start("a.js",19,20,"Dog","function Dog() {");
      return (function(){ var sjsp__return = Dog.__super__.constructor.apply(this, arguments); sjsp__end(sjsp__state); return sjsp__return; } ).call(this);; sjsp__end(sjsp__state);
    }

    return (function(){ var sjsp__return = Dog; sjsp__end(sjsp__state); return sjsp__return; } ).call(this);; sjsp__end(sjsp__state);

  })(Animal);

  o = new Dog({
    name: 'cinnamon'
  });

  console.log(o.name);; sjsp__end(sjsp__state);
```

``` html
<!DOCTYPE html>
<html>
<head>
  <script src="a.sjsp.js" type="text/javascript"></script>
</head>
</html>
```

> Uncaught TypeError: Cannot read property 'name' of undefined 
> a.sjsp.js:9

**Comment by laiso** [comment]
I have a proposition!

**sjsp__arguments**

``` js
function Dog() { 
  var sjsp__state = sjsp__start("a.js",19,20,"Dog","function Dog() {");
  var sjsp__arguments = arguments;
  return (function(){ 
      var sjsp__return = Dog.__super__.constructor.apply(this, sjsp__arguments);
      sjsp__end(sjsp__state);
      return sjsp__return;
      } ).call(this);
  ;
  sjsp__end(sjsp__state);
}
```

**Comment by itchyny** [comment]
This is a serious problem and thank you for your proposition. But rewiting the variable will be a hard work.

Does the following code work well?

``` javascript
function Dog() { 
  var sjsp__state = sjsp__start("a.js",19,20,"Dog","function Dog() {");
  return (function(arguments){      // here pass `arguments`
      var sjsp__return = Dog.__super__.constructor.apply(this, arguments);   // not touch
      sjsp__end(sjsp__state);
      return sjsp__return;
      } ).call(this, arguments);     // here
  ;
  sjsp__end(sjsp__state);
}
```

**Comment by laiso** [comment]
it's perfect :100: 
I believe your idea is the right way.

**Comment by itchyny** [comment]
I fixed the problem. Please pull the repository, rebuild sjsp and confirm.

**Comment by laiso** [comment]
LGTM
I already saw that this problem has been fixed on 250bb6805686eaf97af462534898ce4beef8d723 .

**Comment by itchyny** [comment]
Thanks for your issue report.

(https://github.com/itchyny/sjsp/commit/023180e4b9e32ff3ccfdaf5118d7ac52d653f4b5 is the corresponding commit)
